### PR TITLE
feat: add deal test code

### PR DESCRIPTION
**Pull request by gyuguen**
### More details
* Added MockStore
* Added deal's `validate_data_handler` test code
* Change `requester_address` value from `request.From` to `request.Query`

### CODE REVIEW

File: server/service/datadeal/validate_data_handler_test.go [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+}
+
+func setDefaultURLVars(req *http.Request) *http.Request {
+	pathMap := make(map[string]string)
+	pathMap[types.DealIDKey] = "1"
+	return mux.SetURLVars(req, pathMap)
+}
+
+func TestHandlerValidateData(t *testing.T) {
+	req := httptest.NewRequest(http.MethodGet, "/v0/data-deal/deals/1/data?requester_address="+requesterAddress, bytes.NewReader(defaultData))
```
Comment (hansol-medi) [inline server/service/datadeal/validate_data_handler_test.go:114]: The method might be POST I guess.

File: server/service/datadeal/validate_data_handler_test.go [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+	require.Equal(t, "failed to read query parameter\n", recorder.Body.String())
+}
+
+func TestHandlerValidateDataEmptyBody(t *testing.T) {
+	req := httptest.NewRequest(http.MethodGet, "/v0/data-deal/deals/1/data?requester_address="+requesterAddress, nil)
+	req = setDefaultURLVars(req)
+
+	req.Header.Set("Content-Type", "application/json")
+	req.Form = make(map[string][]string)
+	req.Form.Set("requester_address", requesterAddress)
```
Comment (hansol-medi) [inline server/service/datadeal/validate_data_handler_test.go:187]: When you make a new request, the requesterAddress is given to the request, but it seems to be given again here.

File: server/service/datadeal/validate_data_handler_test.go [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+
+	dealService := makeDataDealService()
+
+	dealService.handleValidateData(recorder, req)
+
+	res := recorder.Result()
+
+	require.Equal(t, http.StatusForbidden, res.StatusCode)
+	require.Equal(t, "JSON schema validation failed\n", recorder.Body.String())
+}
```
Comment (hansol-medi) [inline server/service/datadeal/validate_data_handler_test.go:246]: This test seems to be the same with the above test, `TestHandlerValidateDataInvalidJsonType()`

**Review by hansol-medi — COMMENTED** [review]
Thank you for your works!
Left some comments

**Review by inchori — APPROVED** [review]
LGTM 👍

**Comment by gyuguen** [comment]
> Thank you for your works! Left some comments

Thank you. I fixed it according to your comment.
(https://github.com/medibloc/panacea-oracle/pull/86/commits/b2513ade097ad424eb35e8c151c0756ac15c94c3)

**Review by hansol-medi — APPROVED** [review]
Thx! LGTM

**Pull request merged**